browser/brave_rewards/rewards_tab_helper.cc

```diff
@@ -192,13 +205,43 @@ void RewardsTabHelper::MaybeSavePublisherInfo() {
   }
 
   // The Rewards system currently assumes that the |publisher_info| table is
-  // populated by calling `GetPublisherActivityFromUrl` as the user nativates
+  // populated by calling `GetPublisherActivityFromUrl` as the user navigates
   // the web. Previously, this was accomplished within the background script of
   // the Rewards extension.
   rewards_service_->GetPublisherActivityFromUrl(
       tab_id_.id(), GetWebContents().GetLastCommittedURL().spec(), "", "");
 }
 
+void RewardsTabHelper::OnCreatorDetected(
+    std::optional<CreatorDetectionScriptInjector::Result> result) {
+  if (!rewards_service_ || !result) {
+    return;
+  }
+
+  SetPublisherIdForTab(result->id);
```

Comment:
Thanks for catching this! I've updated this callback to check that the URL still matches.

Reply:
do we want to completely ignore the result in this case and not send anything to the rewards service or do we just not want to update the publisher? Won't we lose the publisher data below
```
// When a creator has been detected for the current tab, we must send the
    // creator data to the utility process so that the "publisher_info" database
    // table can be populated
```

Reply:
I think that if the URL changes before the detection callback is run, then I think it's OK to treat it as if the creator info could not be detected for that previous page load.

Reply:
I feel like we should do something more robust here than checking the url like having a request id that we check.

Reply:
I've updated the check to use `last_detection_url_`, and I think that this is sufficient.

There may be the possibility that a user quickly visits urls `A -> B -> A`, and then we get a detection result for the first `A` and interpret it as a result for the second `A`. The detection result, however, will be valid for either visit.

Reply:
I'm not so much worried about the A -> B -> A thing, it's just not a great way to handle these kinds of things and I want to discourage people from tracking based on the url. I didn't think about it  before, but there is something already here we can use `NavigationHandle::GetGlobalRequestID`